src/udp_packets.rs

```diff
@@ -0,0 +1,277 @@
+
+        // Client hardware address padding: 00000000000000000000
+        packet.extend(&[0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]);
+
+        // Server host name not given
+        packet.extend(&[0x00; 64]);
+
+        // Boot file name not given
+        packet.extend(&[0x00; 128]);
+
+        // Magic cookie: DHCP
+        packet.extend(&[0x63, 0x82, 0x53, 0x63]);
+
+        // DHCP Message Type: DHCP Discover (1)
+        packet.extend(&[0x35, 0x01, 0x01]);
+
+        // DHCP Option: Parameter Request List
+        packet.extend(&[0x37, 0x04, 0x01, 0x03, 0x06, 0x2a]);
+
+        // End Option
+        packet.push(0xff);
+
+        packet
+    }
+
+    /// Constructs a DNS query packet.
+    ///
+    /// The structure of the DNS packet is as follows:
+    ///
+    /// ```text
+    /// +---------------------+
+    /// |        Header       |
+    /// +---------------------+
+    /// |       Question      | the question for the name server
+    /// +---------------------+
+    /// |        Answer       | RRs answering the question
+    /// +---------------------+
+    /// |      Authority      | RRs pointing toward an authority
+    /// +---------------------+
+    /// |      Additional     | RRs holding additional information
+    /// +---------------------+
+    /// ```
+    ///
+    /// The function builds a DNS query packet with the following fields:
+    /// - **Header**: Contains the transaction ID and flags.
+    /// - **Question**: Specifies the query (e.g., the domain name to look up).
+    /// - **Answer**: Resource records that answer the question (empty for a query).
+    /// - **Authority**: Resource records that point toward an authority (empty for a query).
+    /// - **Additional**: Resource records holding additional information (empty for a query).
+    ///
+    /// # Returns
+    ///
+    /// A `Vec<u8>` representing the DNS query packet.
+    pub fn craft_dns_query_packet() -> Vec<u8> {
+        let mut packet = Vec::new();
+
+        // Transaction ID: Random
+        packet.extend(&[0x12, 0x34]);
+
+        // Flags: Standard query
+        packet.extend(&[0x01, 0x00]);
+        // Header ^
+
+        // Questions: 1
+        packet.extend(&[0x00, 0x01]);
+
+        // Answer RRs: 0
+        packet.extend(&[0x00, 0x00]);
+
+        // Authority RRs: 0
+        packet.extend(&[0x00, 0x00]);
+
+        // Additional RRs: 0
+        packet.extend(&[0x00, 0x00]);
+
+        // Query: www.google.com
+        let query_name = "www.google.com";
+        for part in query_name.split('.') {
+            packet.push(part.len() as u8);
+            packet.extend(part.as_bytes());
+        }
+        packet.push(0); // End of query name
+
+        // Query Type: A (host address)
+        packet.extend(&[0x00, 0x01]);
+
+        // Query Class: IN (internet)
+        packet.extend(&[0x00, 0x01]);
+
+        packet
+    }
+
+    /// Creates the simplest ntp packet possibl
```

Comment:
nitpick: casing/typo.

Should be `/// Creates the simplest NTP packet possible`…?